crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -445,56 +550,85 @@ def proposal_update
+  add_help(:proposal_create, [:name], [:put])
+  def proposal_create
+    code, message = @service_object.proposal_create(
+      params.slice(
+        :id,
+        :description,
+        :attributes,
+        :deployment
+      )
+    )
+
     respond_to do |format|
-      format.html {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render text: answer[1]
-      }
-      format.xml  {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render xml: answer[1]
-      }
-      format.json {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render json: answer[1]
-      }
+      case code
+      when 200
+        format.json do
+          render json: message
+        end
+        format.html do
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:controller],
+              id: params[:id]
+            )
+          )
+        end
+      else
+        format.json do
+          render json: { error: message }, status: code
+        end
+        format.html do
+          flash[:alert] = message
+
+          redirect_to(
+            barclamp_modules_path(
+              id: params[:controller]
+            )
+          )
+        end
+      end
     end
   end
 
   #
-  # Provides the restful api call for
-  # Commit Proposal Instance 	/crowbar/<barclamp-name>/<version>/proposals/commit/<barclamp-instance-name> 	POST 	This action will create a new instance based upon this proposal. If the instance already exists, it will be edited and replaced
+  # Currently, A UI ONLY METHOD
   #
-  add_help(:proposal_commit,[:id],[:post])
-  def proposal_commit
-    ret = @service_object.proposal_commit params[:id]
-    return render text: ret[1], status: ret[0] if ret[0] >= 210
-    render json: ret[1], status: ret[0]
-  end
+  add_help(:proposal_status,[:id, :barclamp, :name],[:get])
+  def proposal_status
+    proposals = {}
+    i18n = {}
+    begin
+      active = RoleObject.active(params[:barclamp], params[:name])
 
-  #
-  # Provides the restful api call for
-  # Dequeue Proposal Instance 	/crowbar/<barclamp-name>/<version>/proposals/dequeue/<barclamp-instance-name> 	DELETE 	This action will dequeue an existing proposal.
-  #
-  add_help(:proposal_dequeue,[:id],[:delete])
-  def proposal_dequeue
-    answer = @service_object.dequeue_proposal params[:id]
-    set_flash(answer, "proposal.actions.dequeue_%s")
-    if answer[0] == 200
-      return render json: {}, status: answer[0]
-    else
-      return render text: flash[:alert], status: answer[0]
+      result = if params[:id].nil?
+        Proposal.all
+      else
+        [Proposal.where(barclamp: params[:barclamp], name: params[:name]).first]
+      end
+      result.each do |prop|
+        prop_id = "#{prop.barclamp}_#{prop.name}"
+        status = (["unready", "pending"].include?(prop.status) or active.include?(prop_id))
```

Comment:
Style/AndOr: Use `||` instead of `or`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)